File: 2026/Company/Company.Web/Modules/ServerTypes/Sales/OperationTypeRow.ts
import { fieldsProxy, getLookup, getLookupAsync } from "@serenity-is/corelib";

export interface OperationTypeRow {
    OperationTypeId?: number;
    OpCode?: number;
    Operation?: string;
    IsActive?: number;
    TenantId?: number;
    InsertUserId?: number;
    InsertDate?: string;
    UpdateUserId?: number;
    UpdateDate?: string;
}

export abstract class OperationTypeRow {
    static readonly idProperty = 'OperationTypeId';
    static readonly isActiveProperty = 'IsActive';
    static readonly nameProperty = 'Operation';
    static readonly localTextPrefix = 'Sales.OperationType';
    static readonly lookupKey = 'Sales.OperationType';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<OperationTypeRow>('Sales.OperationType') }
    static async getLookupAsync() { return getLookupAsync<OperationTypeRow>('Sales.OperationType') }

    static readonly deletePermission = 'Company:Currency:Delete';
    static readonly insertPermission = 'Company:Currency:Modify';
    static readonly readPermission = 'Company:Currency:View';
    static readonly updatePermission = 'Company:Currency:Modify';

    static readonly Fields = fieldsProxy<OperationTypeRow>();
}

File: 2026/Company/Company.Web/Modules/Sales/OperationType/OperationTypeRow.cs
using Company.Administration.Entities;
namespace Company.Sales;

[ConnectionKey("Sales"), Module("Sales"), TableName("[sales].[OperationType]")]
[DisplayName("Operation Type"), InstanceName("Operation Type")]
[ReadPermission(SalesPermissionKeys.Currency.View)]
[ModifyPermission(SalesPermissionKeys.Currency.Modify)]
[DeletePermission(SalesPermissionKeys.Currency.Delete)]
[ServiceLookupPermission("SalesPermission:General")]
[LookupScript("Sales.OperationType", LookupType = typeof(MultiTenantRowLookupScript<>))]
public sealed class OperationTypeRow : LoggingRow<OperationTypeRow.RowFields>, IIdRow, INameRow, IIsActiveRow, IMultiTenantRow
{
    [DisplayName("Operation Type Id"), Column("OperationTypeID"), Identity, IdProperty]
    public int? OperationTypeId { get => fields.OperationTypeId[this]; set => fields.OperationTypeId[this] = value; }

    [DisplayName("Op Code"), NotNull]
    public short? OpCode { get => fields.OpCode[this]; set => fields.OpCode[this] = value; }

    [DisplayName("Operation"), Size(30), NotNull, QuickSearch, NameProperty]
    public string Operation { get => fields.Operation[this]; set => fields.Operation[this] = value; }

    #region Tenant & Activ

    [Insertable(false), Updatable(false)]
    public Int32? TenantId { get => fields.TenantId[this]; set => fields.TenantId[this] = value; }

    public Int32Field TenantIdField { get => fields.TenantId; }

    [NotNull, Insertable(false), Updatable(true)]
    public Int16? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    Int16Field IIsActiveRow.IsActiveField { get => fields.IsActive; }

    #endregion Tenant & Activ

    public class RowFields : LoggingRowFields
    {
        public Int32Field OperationTypeId;
        public Int16Field OpCode;
        public StringField Operation;

        public Int16Field IsActive;
        public Int32Field TenantId;

    }
}